### ISSUE TITLE
alternative naming for Execute method

### ISSUE THREAD

**Issue by frankmeola** [michael-wolfenden/Polly#35]
As i read through the readme I liked the syntax and style of the fluent syntax. When i reached how to run a policy the method __Execute__ felt somewhat jarring. Since we are dealing with policies __Enact__ or __Enforce__ might fit better.

Ex)

    policy.Enforce(()=> MyCode());

Or

    policy.Enact(() => MyCode());

Not a huge deal since someone could probably achieve this with extension methods in their code but I think it would bring the policy metaphor through to the end.

**Comment by michael-wolfenden** [comment]
Frank,

Thanks for your feedback. I agree these are better names, but to change them at this point would be a huge breaking change that would everyone using the library.

I will mark as wont fix and close

**Comment by frankmeola** [comment]
Mike, 

Thanks for taking the time to reply. I apologize for the miscommunication. I had intended that the Enact or  Enforce would be __in addition__ to the Execute and would most likely use Execute to handle the implementation. It would be more of a semantic method overload instead of an parameter method overload :-) That way there is no breaking change and people could use whichever method fit their mental model of a policy.

As expected I was able to accomplish this with an extension method so the different syntax is something the client could opt-in to by adding a using statements:

    public static class PollyExtensions
    {
        public static void Enact(this Policy policy, Action action)
        {
            policy.Execute(action);
        }
    }

I could also understand where having two names for the same method would be confusing to the client not to mention a bigger API surface for you to manage.

Thank you for taking the time to listen to suggestions!
